File: 卡码网/Js/test20.js
//20. 删除重复元素
const rl=require("readline").createInterface({input:process.stdin});
const iter=rl[Symbol.asyncIterator]();
const readline=async ()=>(await iter.next()).value;

async function main(){
  while(line=await readline()){
    const nums=line.trim().split(" ").map(Number);
    if(nums[0]===0){
      console.log("list is empty");
      continue;
    }
    let res="";
    for(let i=1;i<=nums[0];i++){
      if(i!=1) res+=" ";
      res+=nums[i];
    }
    console.log(res);
    res="";
    let numsUnique=Array.from(new Set(nums.slice(1)));
    numsUnique.forEach((item,index)=>{
      if(index!=0) res+=" ";
      res+=item;
    });
    console.log(res);
  }
}

main();

File: 卡码网/C++/test20.cpp
//20. 删除重复元素
#include <iostream>

using namespace std;

int main(){
	int n;
	while(scanf("%d",&n)!=EOF){
		if(n==0){
			printf("list is empty\n");
			continue;
		}
		int nums[n];
		for(int i=0;i<n;i++){
			if(i!=0) printf(" ");
			scanf("%d",&nums[i]);
			printf("%d",nums[i]);
		}
		printf("\n%d",nums[0]);
		for(int i=1;i<n;i++){
			if(nums[i]!=nums[i-1]) printf(" %d",nums[i]);
		}
		printf("\n");
	}
	return 0;
}